geoh5py/shared/concatenation/drillholes_concatenator.py

```diff
@@ -251,30 +161,123 @@ def _truncate_arrays_to_smallest(arrays: list[np.ndarray]) -> list[np.ndarray]:
 
         :param output: The data to pass to structured array.
-        :param object_index_dictionary: The dictionary to get the data names.
-        :param first_name: The name of the first column of the structured array.
+        :param names: The name of the columns of the structured array.
 
         :return: The structured array.
         """
         # create the structured array
-        dtype = [(first_name, "O")]
-
-        for idx, data_name in enumerate(
-            list(
-                object_index_dictionary[list(object_index_dictionary.keys())[0]].keys()
-            )
-        ):
-            type_temp = np.array([output[0, idx + 1]]).dtype
+        dtype = []
+        for idx, data_name in enumerate(names):
+            type_temp = np.array([output[0, idx]]).dtype
             if type_temp.kind in ["S", "U"]:
                 dtype.append((data_name, "O"))
             else:
                 dtype.append((data_name, type_temp))
 
         return np.core.records.fromarrays(output.T, dtype=dtype)
+
+    def _push_values_to_drillhole(
+        self,
+        name: str,
+        values: np.ndarray,
+        index: dict,
+        property_group: str,
+    ):
+        """
+        Push the values to the drillhole.
+
+        :param name: The name of the data to push.
+        :param values: The values to push.
+        :param index: The index to use to map the data
+        :param property_group: the name of the property group.
+        :return:
+        """
+        template = list(list(index.values())[0].keys())[0]
+
+        # prepare the association values
+        associations_values = {
+            name: self.data[name]
+            for name in self.unique_property_group_names[property_group][0]
+        }
+
+        if len(associations_values) == 2:
+            association = "from-to"
+        else:
+            association = "depth"
+
+        for drillhole_uid, indices in index.items():
+            # get the drillhole
+            drillhole: Drillhole = self.workspace.get_entity(  # type: ignore
+                UUID(drillhole_uid.decode("utf-8"))
+            )[0]
+
+            # define the specific associations
+            drillhole_association = []
+            for key, value in associations_values.items():
+                drillhole_association.append(
+                    value[indices[key][0] : indices[key][0] + indices[template][1]]
+                )
+
+            drillhole.add_data(
+                {
+                    name: {
+                        "values": values[
+                            indices[template][0] : indices[template][0]
+                            + indices[template][1]
+                        ],
+                        association: np.array(drillhole_association).T,
+                    },
+                },
+                property_group=property_group,
+            )
+
+    def add_values_on_property_group(
```

Comment:
Suggest we change this to `add_values_on_data_table` and pass a DrillholeGroupTable as argument.